parser.c

```diff
@@ -207,10 +222,19 @@ static int parse_line(char *line, struct librole_graph *G)
         goto libnss_role_parse_line_error;
 
     drop_quotes(&last);
-    result = librole_get_gid(last, &role.gid);
+    result = librole_get_gid(last, &role->gid);
     if (result != LIBROLE_OK)
         goto libnss_role_parse_line_error;
 
+    int gid_index;
```

Comment:
Nitpick: the code is not valid C89 any more (due to mixed code and declarations)